types/jsreport-core/index.d.ts

```diff
@@ -0,0 +1,117 @@
+// Type definitions for jsreport-core 1.5
+// Project: http://jsreport.net
+// Definitions by: taoqf <https://github.com/taoqf>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+// TypeScript Version: 2.3
+
+/// <reference types="node" />
+
+declare namespace JsReport {
+  type Helpers = string | { [fun: string]: (...args: any[]) => any };
+
+  const enum EngineType {
```

Comment:
Any reason why you postfix enums with 'Type'? I mean why you changed `Engine` to `EngineType` for example? I see in typescript [convention style guide](https://github.com/basarat/typescript-book/blob/master/docs/styleguide/styleguide.md#enum) that enums are without postfix. Like `Color` instead of `ColorType`.

Reply:
en.. i think may be we need a base interface called `Engine` and `Recipe`, sorry for that I could not name the interface line 59 and line 61. so, I add Type as the postfix.

Reply:
Ah ok. I didn't notice that. The `jsreport.use` function accepts function or object, so I'm not sure if we need base interface `Engine` or `Recipe` now. Give me a moment to think if we can potentially need that in the future.

```js
/**
 * Manual registration of the extension. Once calling `use` the auto discovery of extensions is turned off if not explicitly
 * turned on.
 * jsreport.use(require('jsreport-jsrender')())
 * @param {Object || Function} extensions
 * @return {Reporter} for chaining
 * @public
 */
Reporter.prototype.use = function (extension) {
  this.extensionsManager.use(extension)
  return this
}
```

Reply:
I think if other people want a engine([dot](http://olado.github.io/doT/) is a good template engine as I know), they may want to know how to. and this interface Engine(if it has a exact type defination) may help them.

Reply:
Registering engines and recipes is a little bit more complicated than your types defines.
The `use` function registers an extension, which is later invoked and can register an engine.
The correct signature of the `use` function is currently as this.

```js
type Extension = (reporter: Reporter, definition: object) => void;

interface ExtensionDefinition {
  options: any;
  main: any;
  directory: string;
}

interface Reporter {
  use(extension: Extension | ExtensionDefinition): Reporter;
}
```

The engine is then registered using
```js
jsreport.use((reporter : JsReport.Reporter) => {
  reporter.extensionsManager.engines.push({
    name: 'handlebars',
    pathToEngine: path.join(__dirname, 'handlebarsEngine.js')
  })
})
```
This is not so convenient at this moment, unfortunately.

Reply:
I see.